### DISCUSSION TITLE
How can i separate `jni____.dll/so` from jar file when gradle build?

### DISCUSSION THREAD

**Discussion by devjeonghwan** [bytedeco/javacpp#567] [Q&A] [answered]
@saudet 

I'm building a multi gradle project.

```
root_project (not java project)
   - sub_project_1
     - plugin: org.bytedeco.gradle-javacpp-build
     - compile: org.bytedeco:javacpp

   - sub_project_2
     - plugin: org.bytedeco.gradle-javacpp-build
     - compile: org.bytedeco:cuda-platform
     - implementation: sub_project_1
     - and append '-cuda' to javacppPlatform for nvcc build
     ext {
        javacppPlatform = javacppPlatform + '-cuda'
     }
```

When i execute gradle command "gradlew sub_project_1:jar" and "gradlew sub_project_1:javacppJar". generated two jar files.

```
1. sub_project_1.jar
2. sub_project_1-windows-x86_64.jar
```

`sub_project_1.jar` not include dll or so files. (only empty `windows-x86_64` directories)
`sub_project_1-windows-x86_64.jar` include platform dll or so files. (`jni___.dll/so` files)

but, When i execute gradle command "gradlew sub_project_2:jar" and "gradlew sub_project_2:javacppJar", generated two jar files.

```
1. sub_project_2.jar
2. sub_project_2-windows-x86_64-cuda.jar
```

`sub_project_2.jar` include platform dll or so files.
`sub_project_2-windows-x86_64-cuda.jar` not include dll or so files. (only empty `windows-x86_64` directories)

I want `sub_project_2` to be built separately just like `sub_project_1`.

**Comment by saudet** [comment, marked as answer]
This is about Gradle JavaCPP? We can set the extension with the javacppPlatformExtension project property, which is missing from this list:
https://github.com/bytedeco/gradle-javacpp/blob/master/src/main/java/org/bytedeco/gradle/javacpp/BuildPlugin.java#L43

**Reply by devjeonghwan** [reply]
Yesh maybe about `Gradle JavaCPP`.

I did another test that new sub project without `cuda` platform prefix. it works as i think.

but i want to use `nvcc` compiler instead of `cl.exe`. 

If I set the `javacppPlatformExtension` property you told me to `cuda`, it uses `cl.exe`.
```gradle
ext {
    javacppPlatformExtension = 'cuda'
}
```
or
```gradle
ext {
    javacppPlatformExtension = '-cuda'
}
```

How can I move `jni___.dll/so` from `____.jar` to `____-windows-x86_64.jar` while using `nvcc` compiler?

**Reply by devjeonghwan** [reply]
I have to do javacpp parse and build using 'nvcc' compiler for my gradle project build. (because i use cuda headers and call kernels)

If I modify 'gradle.build' as below for use `nvcc`

```gradle
ext {
    javacppPlatform = javacppPlatform + '-cuda'
}
```

It works. but, problem is 'jni___.dll' files are included in the (`____.jar`) jar file.

I want separated 'jni___.dll' into '_____-windows-x86-64.jar' files.

If I remove `javacppPlatform + '-cuda'` code, DLLs moved to `____.jar`. (but i have to use nvcc. so i can't remove `javacppPlatform + '-cuda'` code)

**Reply by saudet** [reply]
To change the compiler, we can set the properties, such as shown in this sample for Android:
https://github.com/bytedeco/gradle-javacpp/blob/master/samples/zlib/build.gradle#L35

**Reply by devjeonghwan** [reply]
Thanks
